docs/en/tutorials/getting-started.rst

```diff
@@ -228,33 +227,196 @@ entity definition:
+
+    This method, although very common, is inappropriate for Domain Driven
+    Design (DDD), because in such design, methods should be named according
+    to actual business operations, and entities should be valid anytime.
+
+Here is an example of a simple **anemic model**:
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        <?php
+        class User
         {
-            return $this->id;
+            private $username;
+            private $passwordHash;
+            private $bans;
+
+            public function getUsername(): string
+            {
+                return $this->username;
+            }
+
+            public function setUsername(string $username): void
+            {
+                $this->username = $username;
+            }
+
+            public function getPasswordHash(): string
+            {
+                return $this->passwordHash;
+            }
+
+            public function setPasswordHash(string $passwordHash): void
+            {
+                $this->passwordHash = $passwordHash;
+            }
+
+            public function getBans(): array
+            {
+                return $this->bans;
+            }
+
+            public function addBan(Ban $ban): void
+            {
+                $this->bans[] = $ban;
+            }
         }
 
-        public function getName()
+Here, we avoid all possible logic from the model and only care about injecting
+data into it without validation nor consideration about the object's state.
+
+And as Doctrine ORM is a persistence tool for your domain, the state of an
+object is really important. This is why we recommend using rich models.
+
+**Rich models: Mutators and DTOs**
+
+We recommend using a rich model design and rely on more complex
+mutators, ideally based on DTOs.
+This way, you should **not** use getters nor setters, and instead,
+implement methods that represent the **behavior** of your domain.
+
+For example, when having a ``User`` entity, we could foresee
+the following kind of optimization.
+
+Example of a rich model with proper accessors and mutators:
+
+.. configuration-block::
+
+    .. code-block:: php
+        <?php
+        class User
         {
-            return $this->name;
+            private $banned;
+            private $username;
+            private $passwordHash;
+            private $bans;
+
+            public function toNickname(): string
+            {
+                return $this->username;
+            }
+
+            public function authenticate(string $password, callable $checkHash): bool
+            {
+                return $checkHash($password, $this->passwordHash) && ! $this->hasActiveBans();
+            }
+
+            public function changePass(string $password, callable $hash): void
+            {
+                $this->passwordHash = $hash($password);
+            }
+
+            public function ban(\DateInterval $duration): void
+            {
+                assert($duration->invert !== 1);
```

Comment:
`invert`? O_o That's something I didn't know, thanks! :D

Reply:
TIL :bulb:

Reply:
Yep, we can't create "negative" `DateInterval` objects, so we can use this flag to specify it is 😄